tests/features/workspace.feature

```diff
@@ -0,0 +1,147 @@
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I call workspace registerService method with a valid RegisterServiceRequest
+    Then  I expect to receive an error
+
+Scenario: Calling registerService method with an registered service
+    Given  Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run workspace start method
+    And   I call workspace registerService method for a service that was already registered
+    Then  I expect to receive an error
+
+Scenario: Calling registerService method with an invalid serviceName
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run workspace start method
+    And   I call workspace registerService method with invalid values for <serviceName>
+          |<serviceName> |
+          |''        |
+          |{}        |
+          |{ test: 'test' }|
+          |[]        |
+          |['test']  |
+          |null      |
+          |undefined |
+          |true      |
+          |false     |
+          |0         |
+          |-1        |
+    Then  I expect to receive an error
+
+Scenario: Calling registerService method with an invalid displayName
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run workspace start method
+    And   I call workspace registerService method with invalid values for <displayName>
+          |<displayName> |
+          |''        |
+          |{}        |
+          |{ test: 'test' }|
+          |[]        |
+          |['test']  |
+          |null      |
+          |undefined |
+          |true      |
+          |false     |
+          |0         |
+          |-1        |
+    Then I expect to receive an error
+
+Scenario: Calling registerService method with an non-existing service
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run workspace start method
+    And   I call workspace registerService method with valid request but non-existing service
+    Then  I expect to receive an error
+
+Scenario: Calling workspace config should return a promise that includes all existing services and components
```

Comment:
We changed a little bit the API for `config`. Now it takes a request that contain `serviceName` and it will return only the config of this specific service (and not all the workspace config).
Please update the case and also add a case where the `serviceName` provided doesn't exist in the workspace.